### PR TITLE
Map NBT classes

### PR DESCRIPTION
**Pull request by maxpowa**
Also mapped a couple small things in TextComponent/TextFormat since they're used directly by NBT pretty printing.

CompoundWrapper -> PersistedState
baseTag -> key,  it's the key that the tag should be stored as in the parent NBTCompound

### CODE REVIEW

File: mappings/net/minecraft/nbt/TagString.mapping [hunk 1]
```diff
@@ -9,6 +9,12 @@ CLASS ib net/minecraft/nbt/TagString
 		ARG 2 tracker
 	METHOD a write (Ljava/io/DataOutput;)V
 		ARG 0 output
+	METHOD a prettyPrint (Ljava/lang/String;I)Lix;
```
Comment (asiekierka) [inline mappings/net/minecraft/nbt/TagString.mapping]: Is prettyPrint a good conventional name to use?

Comment (asiekierka) [reply mappings/net/minecraft/nbt/TagString.mapping]: It's not. I'd like to propose toComponent or toTextComponent, with keeping of the latest discussins in mind.

Comment (maxpowa) [reply mappings/net/minecraft/nbt/TagString.mapping]: toTextComponent felt kinda weird at the time but prettyPrint does too, I'll go with toTextComponent.

**Comment by maxpowa** [comment]
Comments done

File: mappings/net/minecraft/nbt/CompoundWrapper.mapping [hunk 2]
```diff
@@ -11,3 +11,4 @@ CLASS cee net/minecraft/nbt/CompoundWrapper
 		ARG 0 tag
 	METHOD c markDirty ()V
 	METHOD d getDirty ()Z
+	METHOD e getBaseTag ()Ljava/lang/String;
```
Comment (asiekierka) [inline mappings/net/minecraft/nbt/CompoundWrapper.mapping]: I'm not sure if CompoundWrapper is the best name here. Not sure about getBaseTag either. Could someone look into it?

**Comment by maxpowa** [comment]
We still need to figure out if we want `serialize`/`deserialize` | `toNbt`/`fromNbt` | `toCompound`/`fromCompound` etc...

**Comment by asiekierka** [comment]
https://github.com/FabricMC/pomf/issues/172

**Comment by asiekierka** [comment]
But that's for another day. Let's merge.

**Pull request merged**